pyproject.toml

```diff
@@ -38,6 +38,7 @@ dependencies = [
 "protobuf>=5.28.0,<6.0.0",
 "datasets>=2.15.0,<3.0",
 "simpleeval>=0.9.13,<1.0",
+"HFResourceScanner",
```

Comment:
@aluu317 where is this `HFResourceScanner` residing?  HF Trainer already has some internal tools to measure resource, so want to know what is the delta this provides

Reply:
@fabianlim https://github.com/foundation-model-stack/hf-resource-scanner tthat's the scanner.

Could you send me link to the HF trainer internal tool link?

Reply:
@fabianlim this is more of a light weight solution (very little overhead in terms of run time) and also reports richer information than the internal trainer tool like memory breakup. Furthermore, there is more control through this repo to incrementally add new metrics as needed.

Reply:
@aluu317 you can see it via this args for [time](https://github.com/huggingface/transformers/blob/main/src/transformers/training_args.py#L1485) and [TrainerMemoryTracker](https://huggingface.co/transformers/v4.7.0/_modules/transformers/trainer_utils.html). 

So hf-resource-scanner is only for speed right? @kmehant

Reply:
@fabianlim its for time/speed and memory as well. There is more fine-grained information like what is time taken for forward pass etc.

Reply:
@kmehant yes i noticed after i posted my original message, but I have edited it to ask a differnet thing now, please see

Reply:
@fabianlim Scanner is more general purpose (not just time and memory, but other things as needed) and lightweight in that it instruments for a single target step (not the whole run). 

In general, it is not targeted for prod (though it can be) - mainly for testing/debugging etc.

Reply:
@ChanderG ok then all the more it should not be the default behavior (cc. @aluu317 )

Do you mind sharing some thoughs on my comment on [torch.cuda.](https://github.com/foundation-model-stack/fms-hf-tuning/pull/397#discussion_r1861521569)

Reply:
Not sure exactly what the HFTrainer version is doing, but I have benchmarked Scanner overheads and they are negligible.

Reply:
@ChanderG it depends on model. for the larger ones that consume alot of memory in the GPU I have observed them to be not. In particular when doing `multi-gpu` training.